### PR TITLE
ci: Fix linux performance

### PR DESCRIPTION
**Pull request by juan-lunarg**
1.) Fix ccache misses due to not using matrix.os
2.) Only run 1 job for ubuntu-20.04

### CODE REVIEW

File: .github/workflows/vvl.yml [hunk 1]
```diff
@@ -172,7 +176,7 @@ jobs:
             external/${{ matrix.config }}/Vulkan-Headers/registry
             external/${{ matrix.config }}/SPIRV-Headers/include
             external/${{ matrix.config }}/mimalloc/build/install
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ matrix.arch }}-${{ matrix.os }}-${{ matrix.config }}-${{ hashfiles('scripts/known_good.json') }}
+          key: ${{ matrix.os }}-build-${{ env.cache-name }}-${{ matrix.arch }}-${{ matrix.os }}-${{ matrix.config }}-${{ hashfiles('scripts/known_good.json') }}
```
Comment (juan-lunarg) [inline .github/workflows/vvl.yml:179]: This minor issue adds up tremendously. Adding an additional 30-50 minutes for building dependencies.

Comment (juan-lunarg) [reply .github/workflows/vvl.yml:179]: `runner.os` is equal to linux. Not the current runner image. Which meant the ubuntu-20 and ubuntu-22 caches were fighting each other.

Comment (ncesario-lunarg) [reply .github/workflows/vvl.yml:180]: :grimacing:

**Review by ncesario-lunarg — APPROVED** [review]

**Comment by lunarpapillo** [comment]
Shouldn't line 72 also be modified?

**Comment by juan-lunarg** [comment]
> Shouldn't line 72 also be modified?

Yes it should. I've removed all usage of `runner.os`.

**Review by spencer-lunarg — APPROVED** [review]

**Pull request merged**